### PR TITLE
Allow links to be specified with only the name if it matches the alias

### LINKED ISSUE #11824: Proposal: chance of defining a link as "name" instead of "name:alias" when name and alias are the same
I've seen it's already supported in Docker Compose ([here](https://docs.docker.com/compose/yml/#links)). I think it's more simple and readable.

Example:
```
link: ["mysql", "elasticsearch"]
```
instead of
```
link: ["mysql:mysql", "elasticsearch:elasticsearch"]
```

### PR DESCRIPTION
**Pull request by runcom**
resolves #11824

I think docs still needs something though.

Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

**Comment by LK4D4** [comment]
@runcom Instant fail on `vet` :D 
Change makes sense.

**Comment by runcom** [comment]
@LK4D4 lol yeah I'm fixing vet errors :+1:

**Comment by cpuguy83** [comment]
LGTM

File: docs/man/docker-run.1.md [hunk 1]
```diff
@@ -232,7 +232,8 @@ ENTRYPOINT.
    Read in a line delimited file of labels
 
 **--link**=[]
-   Add link to another container in the form of <name or id>:alias
+   Add link to another container in the form of <name or id>:alias or just <name or id>
+if the alias matches the name
```
Comment (duglin) [inline docs/man/docker-run.1.md]: `in which case the alias will match the name`

**Comment by duglin** [comment]
Just one minor doc tweak but I'll leave that to the editors to decide. 
So LGTM and I'll move to doc review.

File: docs/man/docker-run.1.md (hunk 1 above)
Comment (runcom) [inline docs/man/docker-run.1.md]: thanks!

Comment (jamtur01) [inline docs/man/docker-run.1.md]: This needs to be updated in run.md too I think - https://github.com/docker/docker/blob/master/docs/sources/reference/run.md

**Comment by thaJeztah** [comment]
Docs (so far) look good, but is missing in some places;

- The "linking containers together" user guide (https://docs.docker.com/userguide/dockerlinks/)
- The "docker create" man page (https://github.com/docker/docker/blob/master/docs/man/docker-create.1.md)
- The examples section of the "command line reference" (https://docs.docker.com/reference/commandline/cli/#examples_8), which contains an example on links. I think a short word that the "alias" is optional and an example without alias would be good to include there.

Funny; it looks like the "A complete example" on that page already uses links without alias, which might have been incorrect all along? (Haven't tested it *without* this PR)

**Comment by runcom** [comment]
@thaJeztah updated PTAL

File: docs/sources/reference/commandline/cli.md [hunk 2]
```diff
@@ -2136,6 +2136,12 @@ Guide.
 The `--link` flag will link the container named `/redis` into the newly
 created container with the alias `redis`. The new container can access the
 network and environment of the `redis` container via environment variables.
+The `--link` flag will also just accept the form <name or id> in which case
```
Comment (thaJeztah) [inline docs/sources/reference/commandline/cli.md]: could you add back-ticks around `<name or id>` to prevent Markdown "swallowing" it as a "HTML tag"?

Comment (runcom) [inline docs/sources/reference/commandline/cli.md]: sure, missed that!

Comment (runcom) [inline docs/sources/reference/commandline/cli.md]: updated

**Comment by thaJeztah** [comment]
Docs LGTM, thanks!

/cc @fredlf @jamtur01 @moxiegirl @SvenDowideit

File: docs/sources/userguide/dockerlinks.md [hunk 3]
```diff
@@ -151,6 +151,14 @@ earlier. The `--link` flag takes the form:
 
 Where `name` is the name of the container we're linking to and `alias` is an
 alias for the link name. You'll see how that alias gets used shortly.
+The `--link` flag also takes the form:
+
+	--link <name or id>
+
+in which case the alias will match the name. You could have written the previous
```
Comment (jamtur01) [inline docs/sources/userguide/dockerlinks.md]: In

**Comment by jamtur01** [comment]
Docs LGTM

**Pull request merged**